src/checkstyle/checks.xml

```diff
@@ -32,6 +32,10 @@
         <property name="message" value="Blank line before end of file" />
     </module>
     <module name="RegexpSingleline">
+        <property name="format" value="(class|interface) ([a-zA-Z0-9_])*(<.*>){0,1}(\s)*(extends|implements)" />
```

Comment:
I think the `\s` is wrong since it will match a newline.

You can replace `{0,1}` with `?`.

The `<.*>` works for nested generics because the `*` is greedy and thus will try to match to the last `>` (instead of stopping on the first).